### PR TITLE
[CDAP-8084] CDAP home page - Search, Filtering, Sorting and pagination

### PR DESCRIPTION
**Pull request by tonybach**
JIRA: https://issues.cask.co/browse/CDAP-8084
Build: http://builds.cask.co/browse/CDAP-DRC5518

### CODE REVIEW

File: cdap-ui/package.json [hunk 1]
```diff
@@ -115,6 +115,7 @@
     "react-dropzone": "3.6.0",
     "react-file-download": "0.3.1",
     "react-helmet": "3.1.0",
+    "react-paginate": "^4.1.1",
```
Comment (elfenheart) [inline cdap-ui/package.json]: use specific version for this.

Also I believe you need to rebuild the dll. @ajainarayanan ?

Comment (elfenheart) [reply cdap-ui/package.json]: And I think you might be missing another component as well

Comment (ajainarayanan) [reply cdap-ui/package.json]: Yes. We need to rebuild for this and for `react-timeago` that was added recently.

Comment (tonybach) [reply cdap-ui/package.json]: How do I do that?

File: cdap-ui/app/cdap/components/EntityListView/EntityListInfo/index.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+                containerClassName={"page-list"}
+                activeClassName={"current-page"}
+              />
+            </span>
+        }
+      </span>
+    );
+  }
+  render() {
+    let activeFilters = this.props.activeFilter.map(filter => upperFirst(filter) + 's');
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/EntityListView/EntityListInfo/index.js]: It's probably better for the pluralizations to come from i18n

**Comment by elfenheart** [comment]
might be a good idea to add a `user-select: none` property to the pagination items, and also `outline: 0` on `:focus`

**Comment by elfenheart** [comment]
screen shot 2017-02-07 at 11 46 25 am

The dropdown menu seems to be a little smaller than the dropdown button

Couple of behavior considerations:
- is it possible to change the click event handler for the pagination component to be on the entire  page circle, instead of the text only?
- When you change the filter/sort/search, should we go back to page 1?

**Comment by elfenheart** [comment]
👍  LGTM. Very nice!

**Pull request merged**